src/btree/bt_cursor.c

```diff
@@ -100,16 +99,6 @@ __cursor_page_pinned(WT_CURSOR_BTREE *cbt, bool search_operation)
     if (cbt->ref->page->read_gen == WT_READGEN_OLDEST)
         return (false);
 
-    /*
-     * We need a page with history: updates need complete update lists and a read might be based on
-     * a different timestamp than the one that brought the page into memory. Release the page and
-     * read it again with history if required. Eviction may be locking the page, wait until we see a
-     * "normal" state and then test against that state (eviction may have already locked the page
-     * again).
-     */
-    while ((current_state = cbt->ref->state) == WT_REF_LOCKED)
```

Comment:
Can eviction be locking the page at this point, don't we need to wait till we get a REF_MEM state.

Reply:
Even if we got a `WT_REF_MEM` state, it wouldn't be permanent, that is, it could be flipped back to `WT_REF_LOCKED` after we look at it.

Basically, this is all safe because we're holding a hazard pointer on the page, so even if eviction takes a run at the `WT_REF` and locks it, eviction won't be able to actually evict the page.